src/ol/source/static.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @module ol/source/static
+ */
+
+import {decodeToImageBitmap} from '../Image.js';
+import {getHeight, getWidth} from '../extent.js';
+
+/**
+ * @typedef {Object} LoaderOptions
+ * @property {null|string} [crossOrigin] The `crossOrigin` attribute for loaded images.  Note that
+ * you must provide a `crossOrigin` value if you want to access pixel data with the Canvas renderer.
+ * See https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image for more detail.
+ * @property {import("../extent.js").Extent} [imageExtent] Extent of the image in map coordinates.
+ * This is the [left, bottom, right, top] map coordinates of your image.
+ * @property {import("../size.js").Size} [imageSize] Size of the image in pixels. Usually the image size
+ * is auto-detected, so this only needs to be set for SVG images that have no implicit size.
```

Comment:
I don't think this is useful.  An unsized SVG will still draw to canvas inconsistently and `createImageBitmap` will throw an error if it tries to use it.  Maybe time to deprecate this option.

An alternative to the `load` and `decodeToImageBitmap` methods for SVGs based on #14933 might be a better approach, but perhaps beyond the scope of the PR, something like

```
function svgLoad(image, src) {
  return new Promise((resolve, reject) => {
    let objectUrl;
    function handleLoad() {
      unlisten();
      resolve(image);
    }
    function handleError() {
      unlisten();
      reject(new Error('Image load error'));
    }
    function unlisten() {
      image.removeEventListener('load', handleLoad);
      image.removeEventListener('error', handleError);
      if (objectUrl) {
        URL.revokeObjectURL(objectUrl);
      }
    }
    image.addEventListener('load', handleLoad);
    image.addEventListener('error', handleError);

    if (
      image.crossOrigin !== undefined ||
      src.startsWith('data:image/svg+xml') ||
      location?.origin === new URL(src, location?.origin).origin
    ) {
      fetch(src)
        .then((response) => {
          if (response.headers.get('content-type').includes('image/svg+xml')) {
            return response.text().then((text) => {
              return new Blob([svgDimensions(text)], {type: 'image/svg+xml'});
            });
          }
          return response.blob();
        })
        .then((blob) => {
          objectUrl = URL.createObjectURL(blob);
          image.src = objectUrl;
        })
        .catch(() => {
          image.src = src;
        });
    } else {
      image.src = src;
    }
  });
}
```
(`src` would be required and could not be preset)

Reply:
Happy to remove the `imageSize` option. I only kept it for you, @mike-000.